pkg/replicate/replicator.go

```diff
@@ -183,7 +184,7 @@ func RunReplicate(
 		logger := log.With(logger, "replication-run-id", runID.String())
 		level.Info(logger).Log("msg", "running replication attempt")
 
-		if err := newReplicationScheme(logger, metrics, blockFilter, fetcher, fromBkt, toBkt, reg).execute(ctx); err != nil {
+		if err := newReplicationScheme(logger, metrics, blockFilter, fetcher, fromBkt, toBkt, reg).execute(ctx, concurrencyLevel, reg); err != nil {
```

Comment:
We probably need a validation for the concurrency value.

Reply:
Validation for checking if it is nil?
I have already set the default value for it as "1". 
Is there any other validation that we need to take care of?

Reply:
User can set the value to < 1 so you still need to validate the input value is set properly.